client/ayon_core/tools/attribute_defs/widgets.py

```diff
@@ -459,6 +494,23 @@ def _ui_init(self):
         self.main_layout.addWidget(input_widget, 0)
         self.main_layout.addStretch(1)
 
+        self.setContextMenuPolicy(QtCore.Qt.CustomContextMenu)
+        self.customContextMenuRequested.connect(self._on_context_menu)
+
+    def _on_context_menu(self, pos):
```

Comment:
Menu and action does not have to be stored under `self` they live and die during stack of method.